controller/scheduler/swarm.py

```diff
@@ -1,21 +1,22 @@
 import re
 import time
-
+import logging
 from django.conf import settings
+from docker import errors
 from docker import Client
-
+from api.models import log_event
 from .states import JobState
 
-
 MATCH = re.compile(
     r'(?P<app>[a-z0-9-]+)_?(?P<version>v[0-9]+)?\.?(?P<c_type>[a-z-_]+)?.(?P<c_num>[0-9]+)')
 
 
 class SwarmClient(object):
 
-    def __init__(self, target, auth, options, pkey):
+    def __init__(self, target, auth, options, pkey, app):
```

Comment:
The idea behind the scheduler client is that it's supposed to be a single global client connection for all apps. Instead of adding try/catch blocks here, maybe we should be capturing the APIError and call `log_event` when we call `.create()` in models.py.